File: W1/factorial.c
#include <stdio.h>

int main() {
    int number;
    int factorial = 1;
    
    scanf("%d", &number);
    
    if ( number < 0 ) {
        printf("-1\n");
        return 0;
    }
    for ( int i = 2; i <= number; i++ ) {
        factorial *= i;
    }
    printf("%d\n", factorial);
    
    return 0;
}

File: W5/home_work/factorial.py
def factorial(n):
    if n <= 1:
        if n < 0:
            return -1
        return 1
    
    return n * factorial(n-1)
    

if __name__ == '__main__':
    x = int(input('Enter a number: '))

    print(f'{x}!: {factorial(x)}')
